### DISCUSSION TITLE
API Keys & Hardware Interface changes

### DISCUSSION THREAD

**Discussion by magicbug** [magicbug/Cloudlog#933] [General]
With changes to authentication to improve support for multiple users old design choices have to be fixed (improved) for API keys, and hardware interfaces this will be the following

- Add a User_ID field to each of the tables
- Code changed so that Users only see their own API keys and hardware associated with there account

The result of this change will mean that it will cause old API keys to no longer function, third party applications will not be required to make any changes to their code although users will need to update the applications to use the new keys they generate.

I'm sorry this is a breaking change but it needs to happen in the long term and sooner rather than later.

**Comment by myzinsky** [comment]
When do you plan to push this?

**Reply by magicbug** [reply]
When it's ready, won't be a fit a few more days
